tripy/tripy/common/utils.py

```diff
@@ -21,6 +21,19 @@
 
 from tripy.common.exception import raise_error
 import tripy.common.datatype
+from tripy import export
+
+
+# This is a WAR to avoid implementing general support for recursive type annotations in the registry
+@export.public_api()
+class TensorLiteral:
```

Comment:
Don't you already have recursive type checking though? Seems like it should just work - what's missing?

Reply:
It is not autorecursive. Here is what we would need to check:

```python
tensor_literal = Union[numbers.Number, Sequence["tensor_literal"]]
```

This would be needed to check sequences of arbitrary depth, since you could have a tensor literal of any rank. The `"tensor_literal"` is parsed as a `typing.ForwardRef`. Checking these in the general case can be very, very tricky.

Reply:
I see. Could we put this into a new `types.py` file? We can also document it under a `types` section in API docs.

Reply:
It's been split off into another file.